### ISSUE TITLE
Inconsistent view columns re-ordering causing mismatch between view column lists and SELECT column order

### ISSUE THREAD

**Issue by Simon-Bru** [sqlfluff/sqlfluff#6807]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

We have views defined with below syntax: 
`create or replace view {MY_VIEW} ( col1, col2, col3, col4) as SELECT col1, col2, case when condition then col3 else col5, col4`

When running sql-fluff it gets linted into: 
`create or replace view {MY_VIEW} (col1, col2, col3, col4) as SELECT col2, col1, col4, case when condition then col3 else col5`

This cause some columns misalignment between view definition and selection.

### Expected Behaviour

I would expect the linter to raise an alert for the view definition, and to remove the definition in case it would change the selected columns ordering. 

### Observed Behaviour

Running sql-fluff breaks the view columns ordering, results in the view's `col3` to expose actual values of `col4`

`create or replace view {MY_VIEW} (col1, col2, col3, col4) as SELECT col2, col1, col4, case when condition then col3 else col5`

### How to reproduce

Create a view with a case when statement in the first selected column, and run sql-fluff fix 

### Dialect

Snowflake

### Version

3.3.1

### Configuration

[sqlfluff]
dialect = snowflake
large_file_skip_byte_limit = 0
max_line_length = 110
processes = -1

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = upper
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower

[sqlfluff:rules:convention.not_equal]
preferred_not_equal_style = c_style
[sqlfluff:rules:references.keywords]
ignore_words = account,owner

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by Simon-Bru** [comment]
I got proposed this plugin by GPT which I can't make it work, any help into integrating it would be appreaciated ... 

```python
from sqlfluff.core.rules.base import BaseRule, LintResult
from sqlfluff.core.rules.context import RuleContext
from sqlfluff.core.rules.doc_decorators import document_groups

@document_groups
class Rule_L999(BaseRule):
    """Disallow explicit column list in CREATE VIEW statements."""

    def _eval(self, context: RuleContext) -> LintResult | None:
        if context.segment.type == "create_view_statement":
            # Loop through subsegments to find bracketed column list
            for segment in context.segment.segments:
                if segment.type == "bracketed":
                    return LintResult(
                        anchor=segment,
                        description="Avoid specifying column list in CREATE VIEW; rely on SELECT clause instead."
                    )
        return None

```
